taxonomy-editor-frontend/src/pages/editentry/ListAllProperties.jsx

```diff
@@ -38,23 +49,56 @@ const ListAllProperties = ({ nodeObject, setNodeObject }) => {
                 }}
                 defaultValue={nodeObject.preceding_lines.join('\n')} 
                 variant="outlined" />
-            <Typography sx={{ml: 4, mt: 2, mb: 1}} variant='h5'>
-                Properties
-                <IconButton color="inherit" onClick={handleAdd}>
-                    <AddBoxIcon style={{marginLeft: 6, position: 'relative'}} />
-                </IconButton>
-            </Typography>
-            <div style={{ height: 400, width: '50%' }}>
+            <MaterialTable
+                title={<Typography sx={{ml: 2, mt: 2, mb: 1}} variant='h5'>Properties</Typography>}
+                data={data}
+                columns={[
+                    { title: 'Name', field: 'property-name' },
+                    { title: 'Value', field: 'property-value' }
+                ]}
+                editable={{
+                onRowAdd: (newRow) => new Promise((resolve, reject) => {
+                    const updatedRows = [...data, { id: Math.floor(Math.random() * 100), ...newRow }]
+                    setTimeout(() => {
+                    setData(updatedRows)
+                    resolve()
+                    }, 2000)
+                }),
+                onRowDelete: selectedRow => new Promise((resolve, reject) => {
```

Comment:
mmm, from my understanding this may be better to keep it in a different PR as this one is already super big. Also, do you want to store those changes in the database?, if not I don't see a lot of value in having a delete update, etc

Reply:
@nobeeakon Yep, all the changes need to be tracked, so that I'll be able to update it in the database.

Reply:
still, please if possible, do not continue adding features in this PR

Reply:
amm, you may want to do the databse update and the client update in a single PR so you have the whole feature, as that may  affect the implementation

Reply:
Yes @nobeeakon, after successful implementation of CRUD operations in the table present in `ListAllProperties.jsx`, we can merge after your approval. You have also been invited to the taxonomy-editor team. Hence after accepting it, you would be able to merge PRs too.